modules/tinymce/src/core/test/ts/browser/EditorRemoveTest.ts

```diff
@@ -29,6 +34,51 @@ UnitTest.asynctest('browser.tinymce.core.EditorRemoveTest', (success, failure) =
       cRemoveEditor
     ])),
 
+    Logger.t('remove editor after stylesheets load', Chain.asStep({}, [
+      Chain.op(() => {
+        // setup monitoring for uncaught errors
+        uncaughtErrors = [];
+        window.addEventListener('unhandledrejection', recordError);
+      }),
+      Chain.control(
+        McEditor.cFromHtml('<textarea></textarea>', {
+          ...settings,
+          setup: (editor: Editor) => {
+            editor.on('LoadContent', () => {
+              // Hook the function called when stylesheets are loaded
+              // so we can destroy the editor right after starting to load them.
+              const realLoadAll = editor.ui.styleSheetLoader.loadAll;
+              editor.ui.styleSheetLoader.loadAll = (...args) => {
+                realLoadAll.apply(editor.ui.styleSheetLoader, args);
+                editor.destroy();
+              };
+            });
+          }
+        }), (f, value, next, die, logs) => {
+          f(value, (value2, logs2) => {
+            die('Expected editor would not load completely', logs2);
+          }, (err, logs2) => {
+            // As we have deliberately removed the editor during the loading process
+            // we have to intercept the error that is thrown by McEditor.cFromHtml.
+            if (err === McEditor.errorMessageEditorRemoved) {
+              next(value, logs2);
+            } else {
+              die(err, logs2);
+            }
+          }, logs);
+        }
+      ),
+      Chain.wait(50), // to allow the stylesheet loading to finish
```

Comment:
We should be able to wait for these to be in the DOM shouldn't we? Generally wait's like this are avoided, unless there's no alternative. So could you please see if you can use something like a `UiFinder.cWaitFor(...)` here?

Reply:
What would I be waiting for?

Reply:
The stylesheet to exist in the DOM, so you'd be looking for something like `link[href="..."]`

Reply:
But that doesn't mean that the load event has fired which is what I am waiting for.

Reply:
Essentially, the error occurred after stylesheets had loaded and some code had run to attempt to set the cursor position (maybe the comment is incorrect in this way). If I only wait until the stylesheets exist then I may not catch the error.

Reply:
I suppose what I really want is to wait until there is no asynchronous code still attempting to load or modify the editor.

Reply:
Then you should be able to use a waiter for the stylesheet to appear in `document.styleSheets` instead.

Reply:
It's not a massive issue, but it'd be better to have something that's not potentially going to fail if the load takes more than 50ms.

Reply:
The other problem with waiting on the existence of stylesheets via polling is that they are removed as part of the clean-up of the editor so I could potentially miss them. I could use a mutation observer I suppose but it would have to be added before creating the editor and then I'd be looking for the stylesheets being added and then removed. I can see this getting complex!

Though I do admit that it's hard to know how much waiting is needed to catch the error. On my local machine in Firefox 50ms worked fine however it's easy to imagine IE being way slower than that on the test server.

Reply:
Fair, I guess this should be fine then.